tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5739,9 +5749,7 @@ void CGMSHLSLRuntime::EmitHLSLOutParamConversionCopyBack(
           // Don't need cast.
         } else if (ToTy->getScalarType() == FromTy->getScalarType()) {
           if (ToTy->getScalarType() == ToTy) {
-            DXASSERT(FromTy->isVectorTy() &&
-                         FromTy->getVectorNumElements() == 1,
-                     "must be vector of 1 element");
+            DXASSERT(FromTy->isVectorTy(), "must be vector");
```

Comment:
This is the conversion of the parameter to the out variable. It already truncates correctly and the warning is handled elsewhere while this assert is constrained since it was firing for valid HLSL code.

Reply:
Note: I don't think truncating cast on out param is not supported by fxc, though it should be caught with something other than an assert if so, of course.